app/src/main/java/org/xwiki/android/sync/activities/BaseViewFlipper.kt

```diff
@@ -18,75 +18,66 @@
  * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
  */
 
-package org.xwiki.android.sync.activities;
+package org.xwiki.android.sync.activities
 
-import android.content.Context;
-import android.view.View;
-import androidx.annotation.NonNull;
-import androidx.annotation.Nullable;
-import org.xwiki.android.sync.auth.AuthenticatorActivity;
+import android.content.Context
+import android.view.View
+import org.xwiki.android.sync.auth.AuthenticatorActivity
 
 /**
  * Base class for any flipper which will be used in authenticator.
  *
- * @version $Id$
+ * @version $Id: f3c3c73d439018e529af43268d6d8c807d59f06e $
  */
-public abstract class BaseViewFlipper {
-
+abstract class BaseViewFlipper
+/**
+ * All childs of this class must use only this constructor!
+ *
+ * @param activity Current activity
+ * @param contentRootView Root view of current flipper
+ */
+    (
     /**
      * Activity for operations.
      */
-    protected AuthenticatorActivity mActivity;
-
+    protected var mActivity: AuthenticatorActivity,
     /**
-     * Context of operations executing. In fact, represent {@link #mActivity}.
+     * Root view of flipper.
      */
-    protected Context mContext;
+    protected var mContentRootView: View
+) {
 
     /**
-     * Root view of flipper.
+     * Context of operations executing. In fact, represent [.mActivity].
      */
-    protected View mContentRootView;
+    protected var mContext: Context
 
     /**
-     * All childs of this class must use only this constructor!
-     *
-     * @param activity Current activity
-     * @param contentRootView Root view of current flipper
+     * @return Title of flipper or null if have no
      */
-    public BaseViewFlipper(
-        @NonNull AuthenticatorActivity activity,
-        @NonNull View contentRootView
-    ) {
-        mActivity = activity;
-        mContext = mActivity;
-        mContentRootView = contentRootView;
+    val title: String?
```

Comment:
@jaindiv26 can you, please, explain meaning of this property? It will always be null and have no backing field. So, how can it be used?

Reply:
In AuthenticatorActivity it used like this:

  var title = flipper.title
        if (title == null) {
            title = getString(R.string.app_name)
        }

So in the baseViewFlipper we're just initializing the title with a null value. And in authenticator setting a new value altogether.

Reply:
in this case will be better to remove this property and replace the lines from your example by:

`val title = getString(R.string.app_name)`

Because this property (`title`) have no backing fields and actually final. It is impossible to change the value of this property in class.